actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -337,7 +337,7 @@ def reset_session
       else
         self.session = {}
       end
-      self.flash = nil
+      reset_flash
```

Comment:
Shouldn't the flash middleware just override `reset_session`, call super and reset the flash?